src/components/profile.ts

```diff
@@ -0,0 +1,169 @@
+import { openDB } from './db';
+
+// since interfaces are for compile time there is no easy way to extract keys
+// so this userProfileKeys is needed
+export const userProfileKeys = [
+  'about_me',
+  'birth_date',
+  'preferred_name',
+  'preferred_pronouns',
+  'term',
+  'year',
+  'faculty',
+  'program',
+  'specialization'
+];
+
+export interface UserProfile {
+  about_me?: string;
+  birth_date?: string;
+  preferred_name?: string;
+  preferred_pronouns?: string;
+  term?: string;
+  year?: number;
+  faculty?: string;
+  program?: string;
+  specialization?: string;
+}
+
+const validFaculties = ['Mathematics', 'Engineering', 'Arts', 'Health', 'Science'];
+const validTerms: string[] = ['1A', '1B', '2A', '2B', '3A', '3B', '4A', '4B', 'MASc', 'PHD', 'Alumni', 'Professor'];
+const yearStart = 1900;
+const yearEnd = new Date().getFullYear() + 7;
+
+export enum configMaps {
+  aboutme = 'about_me',
+  birthdate = 'birth_date',
+  preferredname = 'preferred_name',
+  preferredpronouns = 'preferred_pronouns',
+  term = 'term',
+  year = 'year',
+  faculty = 'faculty',
+  program = 'program',
+  specialization = 'specialization'
+}
+
+const validBirthdates = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December'
+];
+
+export const customizationLimits = {
+  aboutme:
+    'Feel free to enter anything you want about yourself! Your hobbies, interests, etc. Must be less than 1000 characters.',
+  birthdate:
+    'To protect your personal information, this must only be your month. Must be one of ' + validBirthdates.join(', '),
```

Comment:
only be your month **of birth**
but eh I guess it's fine as is

Reply:
I can change that when I get home, will merge once alex approves